python-stdlib/gzip/gzip.py

```diff
@@ -1,29 +1,28 @@
-# import zlib
-import uzlib as zlib
+# MicroPython gzip module
+# MIT license; Copyright (c) 2023 Jim Mussared
 
-FTEXT = 1
-FHCRC = 2
-FEXTRA = 4
-FNAME = 8
-FCOMMENT = 16
+_WBITS = const(15)
+
+import io, deflate
+
+if hasattr(deflate.DeflateIO, "write"):
+
+    def compress(data):
+        f = io.BytesIO()
+        with deflate.DeflateIO(f, deflate.GZIP, _WBITS) as g:
+            g.write(data)
+        return f.getvalue()
 
 
 def decompress(data):
-    assert data[0] == 0x1F and data[1] == 0x8B
-    assert data[2] == 8
-    flg = data[3]
-    assert flg & 0xE0 == 0
-    i = 10
-    if flg & FEXTRA:
-        i += data[11] << 8 + data[10] + 2
-    if flg & FNAME:
-        while data[i]:
-            i += 1
-        i += 1
-    if flg & FCOMMENT:
-        while data[i]:
-            i += 1
-        i += 1
-    if flg & FHCRC:
-        i += 2
-    return zlib.decompress(memoryview(data)[i:], -15)
+    f = io.BytesIO(data)
+    with gzip.GzipFile(f, deflate.GZIP, _WBITS) as g:
```

Comment:
Should this be `deflate.DeflateIO`?

Reply:
It should be GZipFile, but without the extra args. Updated.